java/app/src/main/java/com/fullstorydev/shoppedemo/data/ItemDao.java

```diff
@@ -19,10 +19,10 @@
     @Query("SELECT quantityInCart FROM items WHERE title = :title LIMIT 1")
     protected abstract int getItemQuantityByTitle(String title);
 
-    @Query("SELECT ROUND(IFNULL(SUM(quantityInCart * price),0.00),2) FROM items")
+    @Query("SELECT ROUND(IFNULL(SUM(quantityInCart * price), 0.00), 2) FROM items")
     abstract LiveData<Double> getSubtotal();
 
-    @Query("SELECT SUM(quantityInCart) FROM items")
+    @Query("SELECT IFNULL(SUM(quantityInCart), 0) FROM items")
```

Comment:
I've not used android Rooms - what happened before this fix? would the LiveData never call onChange?

Reply:
The change only handles when there is no item  in the cart. In that case before the change the "count in cart" number next to the cart icon displays "null" rather than "0", this change just defaults it to "0". 
But the LiveData will always call onChange, so if there are things in the cart everything is fine.